From: Stephen Farrell <stephen.farrell@cs.tcd.ie>
Date: Fri, 24 Nov 2023 15:03:43 +0000
Subject: change in external ext name

---
 modules/ssl/ssl_engine_kernel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/ssl/ssl_engine_kernel.c b/modules/ssl/ssl_engine_kernel.c
index 24bff77..fb3f578 100644
--- a/modules/ssl/ssl_engine_kernel.c
+++ b/modules/ssl/ssl_engine_kernel.c
@@ -2537,7 +2537,7 @@ int ssl_callback_ClientHello(SSL *ssl, int *al, void *arg)
 
 #ifdef HAVE_OPENSSL_ECH
 
-    if (SSL_client_hello_get0_ext(ssl, TLSEXT_TYPE_ech13, &pos, &remaining)) {
+    if (SSL_client_hello_get0_ext(ssl, TLSEXT_TYPE_ech, &pos, &remaining)) {
         ap_log_cerror(APLOG_MARK, APLOG_INFO, 0, c, APLOGNO(10243)
                       "there is an ECH extension");
     } else {
