Description: h2 request headers clone instead of copy
Author: Stefan Eissing
Origin: upstream, https://bz.apache.org/bugzilla/attachment.cgi?id=38527&action=diff
Bug: https://bz.apache.org/bugzilla/show_bug.cgi?id=66539
Bug-Debian: https://bugs.debian.org/1033408
Forwarded: not-needed
Applied-Upstream: 2.4.57
Reviewed-By: Yadd <yadd@debian.org>
Last-Update: 2023-04-02

--- a/modules/http2/h2_request.c
+++ b/modules/http2/h2_request.c
@@ -279,7 +279,7 @@
 apr_bucket *h2_request_create_bucket(const h2_request *req, request_rec *r)
 {
     conn_rec *c = r->connection;
-    apr_table_t *headers = apr_table_copy(r->pool, req->headers);
+    apr_table_t *headers = apr_table_clone(r->pool, req->headers);
     const char *uri = req->path;
 
     AP_DEBUG_ASSERT(req->authority);
@@ -303,7 +303,7 @@
 {
     const char *cl;
 
-    r->headers_in = apr_table_copy(r->pool, req->headers);
+    r->headers_in = apr_table_clone(r->pool, req->headers);
     if (req->authority) {
         /* for internal handling, we have to simulate that :authority
          * came in as Host:, RFC 9113 ch. says that mismatches between
