# Revert part of r1824187 which requires a newer mod_proxy
--- apache2.orig/modules/http2/mod_proxy_http2.c
+++ apache2/modules/http2/mod_proxy_http2.c
@@ -385,7 +385,7 @@ run_connect:
     }
     
     /* Step Three: Create conn_rec for the socket we have open now. */
-    status = ap_proxy_connection_create_ex(ctx->proxy_func, ctx->p_conn, ctx->r);
+    status = ap_proxy_connection_create(ctx->proxy_func, ctx->p_conn, ctx->owner, ctx->server);
     if (status != OK) {
         ap_log_cerror(APLOG_MARK, APLOG_DEBUG, status, ctx->owner, APLOGNO(03353)
                       "setup new connection: is_ssl=%d %s %s %s", 
